opendps/func_settings.c

```diff
@@ -88,82 +88,52 @@ static bool select_mode;
  * Get/Set functions used as callbacks for all editable fields
  */
 static int32_t get_v_adc_k(void);
-static void set_v_adc_k(ui_number_t *item);
+static void set_v_adc_k(struct ui_number_t *);
 static int32_t get_v_adc_c(void);
-static void set_v_adc_c(ui_number_t *item);
+static void set_v_adc_c(struct ui_number_t *);
 static int32_t get_v_dac_k(void);
-static void set_v_dac_k(ui_number_t *item);
+static void set_v_dac_k(struct ui_number_t *);
 static int32_t get_v_dac_c(void);
-static void set_v_dac_c(ui_number_t *item);
+static void set_v_dac_c(struct ui_number_t *item);
 static int32_t get_a_adc_k(void);
-static void set_a_adc_k(ui_number_t *item);
+static void set_a_adc_k(struct ui_number_t *item);
 static int32_t get_a_adc_c(void);
-static void set_a_adc_c(ui_number_t *item);
+static void set_a_adc_c(struct ui_number_t *item);
 static int32_t get_a_dac_k(void);
-static void set_a_dac_k(ui_number_t *item);
+static void set_a_dac_k(struct ui_number_t *item);
 static int32_t get_a_dac_c(void);
-static void set_a_dac_c(ui_number_t *item);
+static void set_a_dac_c(struct ui_number_t *item);
 static int32_t get_vin_adc_k(void);
-static void set_vin_adc_k(ui_number_t *item);
+static void set_vin_adc_k(struct ui_number_t *item);
 static int32_t get_vin_adc_c(void);
-static void set_vin_adc_c(ui_number_t *item);
+static void set_vin_adc_c(struct ui_number_t *item);
 static int32_t get_brightness(void);
-static void set_brightness(ui_number_t *item);
+static void set_brightness(struct ui_number_t *);
 static int32_t get_refresh(void);
-static void set_refresh(ui_number_t *item);
+static void set_refresh(struct ui_number_t *item);
 
-/*
- * Field labels
- */
-const char* const field_label[] = {
-    "Scr-LED",
-    "Refresh",
-    "V ADC K",
-    "V ADC C",
-    "V DAC K",
-    "V DAC C",
-    "I ADC K",
-    "I ADC C",
-    "I DAC K",
-    "I DAC C",
-    "Vin ADC C",
-    "Vin ADC K",
-};
 
-/*
- * Field get callbacks
- */
-get_func get_functions[] = {
-    &get_brightness,
-    &get_refresh,
-    &get_v_adc_k,
-    &get_v_adc_c,
-    &get_v_dac_k,
-    &get_v_dac_c,
-    &get_a_adc_k,
-    &get_a_adc_c,
-    &get_a_dac_k,
-    &get_a_dac_c,
-    &get_vin_adc_k,
-    &get_vin_adc_c,
+struct field_item {
+    const char* label;
+    int32_t min;
+    int32_t max;
+    int32_t (*get)(void);
+    void (*set)(struct ui_number_t *i);
 };
 
-/*
- * Field set callbacks
- */
-set_func set_functions[] = {
-    &set_brightness,
-    &set_refresh,
-    &set_v_adc_k,
-    &set_v_adc_c,
-    &set_v_dac_k,
-    &set_v_dac_c,
-    &set_a_adc_k,
-    &set_a_adc_c,
-    &set_a_dac_k,
-    &set_a_dac_c,
-    &set_vin_adc_k,
-    &set_vin_adc_c,
+struct field_item field_items[] = {
+    {"Scr-LED",    10000,  1000000,    &get_brightness,     &set_brightness },
```

Comment:
How about "Brightness" ;)

Reply:
I think it's slightly too long, I'm afraid.

Reply:
I had in my head that everything was monospaced. It's actually fine and I've made the changes in the most recent commit.